### PR TITLE
Merge 2.3.1 to main

### PR DESCRIPTION
**Pull request by treid314**
#### What this PR does
Merges the 2.3.1 patch release to main

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,7 +14,6 @@
 * [FEATURE] Query-scheduler: added an experimental ring-based service discovery support for the query-scheduler. Refer to [query-scheduler configuration](https://grafana.com/docs/mimir/next/operators-guide/architecture/components/query-scheduler/#configuration) for more information. #2957
 * [FEATURE] Introduced the experimental endpoint `/api/v1/user_limits` exposed by all components that load runtime configuration. This endpoint exposes realtime limits for the authenticated tenant, in JSON format. #2864 #3017
 * [FEATURE] Query-scheduler: added the experimental configuration option `-query-scheduler.max-used-instances` to restrict the number of query-schedulers effectively used regardless how many replicas are running. This feature can be useful when using the experimental read-write deployment mode. #3005
-* [ENHANCEMENT] Distributor: Add `cortex_distributor_query_ingester_chunks_deduped_total` and `cortex_distributor_query_ingester_chunks_total` metrics for determining how effective ingester chunk deduplication at query time is. #2713
```
Comment (pstibrany) [inline CHANGELOG.md:17]: This changelog entry was removed by accident.

Comment (pracucci) [reply CHANGELOG.md]: I've added it back: https://github.com/grafana/mimir/pull/3068/commits/4172a1b11898f6d577d83dbaeb5aa416177dbdd7

**Review by pracucci — APPROVED** [review]

**Comment by pstibrany** [comment]
Please don't use squash and merge here. I disabled auto-merge.

**Comment by pracucci** [comment]
> Please don't use squash and merge here. I disabled auto-merge.

Thanks for noticing it so quickly. I always forget about it! Will do a commit merge on green.

File: CHANGELOG.md (hunk 1 above)
Comment (treid314) [inline CHANGELOG.md:17]: This line was duplicated in 2.3.0 https://github.com/grafana/mimir/pull/3068/files/f3de2acda7da018afce21d5ec6448b046f872d3f#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR113

Comment (pracucci) [reply CHANGELOG.md:17]: Ops. I'm fixing it.

Comment (pracucci) [reply CHANGELOG.md:17]: Fix here: https://github.com/grafana/mimir/pull/3075

**Pull request merged**